### PR TITLE
feat(plugin): add building search UI

### PR DESCRIPTION
**Pull request by airslice**
# Overview

This PR adds building search UI.

Building search is a popup of sidebar widget.

_Data/communication part is NOT included in this PR_

## What I've done

- UI of building search popup.

## How I tested

UI preview project [here](https://test.reearth.dev/edit/01gr10f0e34kha1f3hnsgxfcpn). 

This preview project will auto popup when initialize the sidebar.

## Memo

### CODE REVIEW

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    },
+    [handleResize],
+  );
+
+  useEffect(() => {
+    addEventListener("message", onMessage);
+    return () => {
+      removeEventListener("message", onMessage);
+    };
+  }, [onMessage]);
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts:193]: I think if you add anything to the deps array of a useEffect with addEventListener you will get multiple even listeners in the lifetime of the the component. Just move the onMessage function inside the useEffect.

**Review by KaWaite — APPROVED** [review]

File: plugin/src/sidebar.ts [hunk 2]
```diff
@@ -55,6 +56,8 @@ const sidebarInstance: PluginExtensionInstance = reearth.plugins.instances.find(
 
 // ************************************************
 // initializations
+let buildingSearchIsOpen = false;
```
Comment (KaWaite) [inline plugin/src/sidebar.ts:59]: Can you move this up where the other xxIsOpen = false let statements?

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts (hunk 1 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts:193]: Emmm i think even move it inside, the `handleResize` will also become the dependency. I think it's reasonalble to list as dependency and when dependency changes the returned clear function will remove previous event listner so there won't be multiple, right?

File: plugin/src/sidebar.ts (hunk 2 above)
Comment (airslice) [reply plugin/src/sidebar.ts:59]: Sure. Updated.

**Pull request merged**